core/interfaces/src/application.rs

```diff
@@ -79,139 +87,119 @@ pub trait ApplicationInterface<C: Collection>:
-    /// between any two node might not exits, but there does exits a path from `n1` to `n2`
-    /// which may cross any `node_i`, a page rank like algorithm is needed here to measure
-    /// a relative score between two nodes.
-    ///
-    /// Existence of this data can allow future optimizations of the network topology.
-    fn get_relative_score(&self, n1: &NodePublicKey, n2: &NodePublicKey) -> u128;
+    /// Query Reputation Measurements Table
+    /// Returns the reported reputation measurements for a node.
+    fn get_reputation_measurements(
+        &self,
+        node: &NodeIndex,
+    ) -> Option<Vec<ReportedReputationMeasurements>>;
 
-    /// Returns information about a single node.
-    fn get_node_info(&self, id: &NodePublicKey) -> Option<NodeInfo>;
+    /// Query Latencies Table
+    fn get_latencies(&self, node_1: &NodeIndex, node_2: &NodeIndex) -> Option<Duration>;
 
-    /// Returns information about a single node using its index.
-    fn get_node_info_with_index(&self, node_index: &NodeIndex) -> Option<NodeInfo>;
+    /// Query Reputation Scores Table
+    /// Returns the global reputation of a node.
+    fn get_reputation_score(&self, node: &NodeIndex) -> Option<u8>;
 
-    /// Returns information about an account.
-    fn get_account_info(&self, id: &EthAddress) -> Option<AccountInfo>;
+    /// Query Total Served Table
+    /// Returns total served for all commodities from the state for a given epoch
+    fn get_total_served(&self, epoch: &Epoch) -> Option<TotalServed>;
 
-    /// Returns a full copy of the entire node-registry, but only contains the nodes that
-    /// are still a valid node and have enough stake.
-    fn get_node_registry(&self, paging: Option<PagingParams>) -> Vec<NodeInfo>;
+    /// Checks if an transaction digest has been executed this epoch.
+    fn has_executed_digest(&self, digest: TxHash) -> bool;
 
-    /// Same as `get_node_registry` but it includes the node index in the result.
-    fn get_node_registry_with_index(&self, paging: Option<PagingParams>) -> Vec<NodeInfoWithIndex>;
+    /// Get Node's Public Key based on the Node's Index
+    fn index_to_pubkey(&self, node_index: &NodeIndex) -> Option<NodePublicKey>;
 
-    /// Returns true if the node is a valid node in the network, with enough stake.
-    fn is_valid_node(&self, id: &NodePublicKey) -> bool;
+    /// Simulate Transaction
+    fn simulate_txn(&self, txn: TransactionRequest) -> TransactionResponse;
 
-    /// Returns the amount that is required to be a valid node in the network.
-    fn get_staking_amount(&self) -> u128;
+    /// Returns the uptime for a node from the past epoch.
+    fn get_node_uptime(&self, node_index: &NodeIndex) -> Option<u8>;
+}
 
-    /// Returns the randomness that was used to start the current epoch.
-    fn get_epoch_randomness_seed(&self) -> &[u8; 32];
+#[infusion::blank]
+pub trait QueryRunnerExt: Clone + Send + Sync + 'static {
```

Comment:
Why not `QueryRunnerExt: QueryRunner` and blanket impls? This way the `+ QueryRunnerExt` isn't required by any generics and it only needs to be brought into scope and used downstream